modules/table/src/main/java/org/apache/ignite/internal/table/distributed/replicator/PartitionReplicaListener.java

```diff
@@ -1086,10 +1111,30 @@ private CompletableFuture<Void> processTxCleanupAction(TxCleanupReplicaRequest r
 
             return raftClient
                     .run(txCleanupCmd)
-                    .thenRun(() -> releaseTxLocks(request.txId()));
+                    .thenCompose(ignored -> allOffFuturesExceptionIgnored(txReadFutures, request)
+                            .thenRun(() -> releaseTxLocks(request.txId())));
```

Comment:
Today, it is useful, because we can no-wait of read operations to commit write intents.
Possibly, we may not wait read operation at all in the future.